public/js/actions/tests/newSource.js

```diff
@@ -1,14 +1,21 @@
 "use strict";
 
-const { actions, selectors, createStore } = require("../../util/test-head");
+const { constants, selectors, createStore } = require("../../util/test-head");
 const fixtures = require("../../test/fixtures/foobar.json");
-const { newSource } = actions;
 const { getSourceByActor } = selectors;
 const sourcesFixtures = fixtures.sources.sources;
 
 const store = createStore();
 const expect = require("expect.js");
 
+// Write our own `newSource` to bypass the batching logic.
+function newSource(source) {
```

Comment:
I agree, I liked preserving the synchronous nature of action to render. Which this actually still does but delaying the action. 

Are you thinking we'd throttle in connect? That might make it tougher to unit test the component.